src/containers/my-quizzes/create-or-edit-quiz-container/CreateOrEditQuizContainer.tsx

```diff
@@ -1,37 +1,95 @@
 import { useModalContext } from '~/context/modal-context'
+import { useSnackBarContext } from '~/context/snackbar-context'
+import { ResourceService } from '~/services/resource-service'
 import QuestionsList from '~/containers/questions-list/QuestionsList'
 import AddQuestions from '~/containers/my-resources/add-questions/AddQuestions'
 import CreateOrEditQuizQuestion from '~/containers/my-quizzes/create-or-edit-quiz-question/CreateOrEditQuizQuestion'
 import AppButton from '~/components/app-button/AppButton'
 import AppTextField from '~/components/app-text-field/AppTextField'
 import PageWrapper from '~/components/page-wrapper/PageWrapper'
 
+import { snackbarVariants } from '~/constants'
+import { authRoutes } from '~/router/constants/authRoutes'
 import { myResourcesPath } from '~/pages/create-or-edit-lesson/CreateOrEditLesson.constants'
-import { styles } from '~/containers/my-quizzes/edit-quiz-container/EditQuizContainer.styles'
+import { QuizContentProps } from '~/pages/new-quiz/NewQuiz.constants'
+import { styles } from '~/containers/my-quizzes/create-or-edit-quiz-container/CreateOrEditQuizContainer.styles'
 import {
   ButtonTypeEnum,
   ButtonVariantEnum,
+  ErrorResponse,
+  NewQuiz,
   Question,
+  Quiz,
   SizeEnum,
   TextFieldVariantEnum
 } from '~/types'
-import { QuizContentProps } from '~/pages/new-quiz/NewQuiz.constants'
+import {
+  defaultResponse,
+  initialValues
+} from '~/containers/my-quizzes/create-or-edit-quiz-container/CreateOrEditQuizContainer.constants'
 
-const EditQuizContainer = ({ questions, setQuestions }: QuizContentProps) => {
+const CreateOrEditQuizContainer = ({
+  title,
+  setTitle,
+  description,
+  setDescription,
+  questions,
+  setQuestions
+}: QuizContentProps) => {
   const { t } = useTranslation()
+  const { setAlert } = useSnackBarContext()
   const { openModal } = useModalContext()
-  const [title, setTitle] = useState<string>('')
-  const [description, setDescription] = useState<string>('')
+  const navigate = useNavigate()
   const [isCreationOpen, setIsCreationOpen] = useState<boolean>(false)
 
+  const handleResponse = () => {
+    setAlert({
+      severity: snackbarVariants.success,
+      message: 'myResourcesPage.quizzes.successAddedOuiz'
+    })
+    navigate(authRoutes.myResources.root.path)
+  }
+
+  const onResponseError = (error: ErrorResponse) => {
+    setAlert({
+      severity: snackbarVariants.error,
+      message: error ? `errors.${error.message}` : ''
+    })
+  }
+
+  const addQuiz = (): Promise<AxiosResponse> => {
+    const quiz = {
+      ...data,
+      title,
+      description,
+      items: questions.map((question) => question._id)
+    }
+    return ResourceService.addQuiz(quiz)
+  }
+
+  const { fetchData: addNewQuiz } = useAxios<Quiz, NewQuiz>({
+    service: addQuiz,
+    fetchOnMount: false,
+    defaultResponse,
+    onResponse: handleResponse,
+    onResponseError: onResponseError
+  })
+
+  const { data } = useForm<NewQuiz>({
```

Comment:
you don`t need to use useForm hook here, you can just pass title, description and questions from props to addNewQuiz callback

Reply:
Yes, you are right, thanks!